### DISCUSSION TITLE
Upgrade the base Alpine OS

### DISCUSSION THREAD

**Discussion by Sangeppato** [abiosoft/colima#585] [Q&A]
How should we upgrade the base Alpine Linux image?
It looks like running the usual `apk update`/`apk upgrade` is not working in a persistent way (I guess `tmpfs` is used?), so should we delete the instance and create a new one from time to time to upgrade Alpine?
Or can we simply replace `basedisk` by downloading a new image from `abiosoft/alpine-lima`?

**Comment by abiosoft** [comment]
To be honest, the VM is not designed to be updated and only selected directories persist across reboots.

If you are not particular about Alpine, you can use the `--layer` flag to enable Ubuntu and use that instead. Whatever you do in the Ubuntu environment persist across reboots.

**Reply by Sangeppato** [reply]
Thanks!
In case you release a new Alpine image and I want to upgrade an older VM I have, can I just replace the `basedisk`? Or should I backup my Docker data volumes, delete the VM and create a new one?

**Reply by abiosoft** [reply]
That can actually work but I have not tested it.

**Reply by Sangeppato** [reply]
I've tried to downgrade `basedisk` and the machine still boots just fine, but was not actually downgraded (apparently many files are copied to `diffdisk` during initialization), so it looks like the only way to upgrade the base OS is to delete and re-create the VM (losing all the images and volumes).

@abiosoft
Since Lima now supports additional data disks, do you think it would be possible for Colima to use a disk mounted at `/var/lib/docker`, in order to persist Docker data across different VM initializations?

**Reply by abiosoft** [reply]
@Sangeppato that's a good point. It will definitely be considered.

Thanks.

**Reply by Sangeppato** [reply]
> @Sangeppato that's a good point. It will definitely be considered.
> 
> Thanks.

That's great, thank you!

FYI, it looks like `finch` does [something similar](https://github.com/runfinch/finch/pull/133) for `containerd`, maybe this could be useful
